### PR TITLE
router/jwt_authn: remove support for v2 deprecated regex and cors.

### PR DESCRIPTION
**Pull request by htuch**
Risk level: Low
Testing: Unit tests modified.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: source/common/http/header_utility.cc [hunk 1]
```diff
@@ -41,12 +41,6 @@ HeaderUtility::HeaderData::HeaderData(const envoy::config::route::v3::HeaderMatc
     header_match_type_ = HeaderMatchType::Value;
     value_ = config.exact_match();
     break;
-  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::
```
Comment (alyssawilk) [inline source/common/http/header_utility.cc:44]: I don't love that there's a default in this switch statement.  lazy check: we'll fail to load before we can hit tihs case yeah?

**Review by alyssawilk — APPROVED** [review]
YAY!!!!

LGTM modulo that one question

**Comment by alyssawilk** [comment]
oh, and needs merge :-/

File: source/common/http/header_utility.cc (hunk 1 above)
Comment (yanavlasov) [reply source/common/http/header_utility.cc:44]: We could add LOG_EVERY_N error when the default is hit.

Comment (htuch) [reply source/common/http/header_utility.cc:44]: This is intentional behavior it seems, see https://github.com/envoyproxy/envoy/blob/9785c3d08c30612a7161c9b7bba9772281f7c6f7/api/envoy/config/route/v3/route_components.proto#L1846

**Comment by htuch** [comment]
Conflict fixed, PTAL and please merge if looks good as I'm OOO. Thanks.

**Review by alyssawilk — APPROVED** [review]
LGTM!!!!

**Comment by alyssawilk** [comment]
cc @phlax just to make sure this one was on your radar

**Pull request merged**